src/Modules/Content/Dnn.PersonaBar.Pages.Tests/RestorePageUnitTests.cs

```diff
@@ -0,0 +1,127 @@
+﻿using NUnit.Framework;
+using Moq;
+using Dnn.PersonaBar.Library.Prompt;
+using Dnn.PersonaBar.Recyclebin.Components;
+using Dnn.PersonaBar.Recyclebin.Components.Prompt.Commands;
+using DotNetNuke.Entities.Portals;
+using DotNetNuke.Entities.Tabs;
+using Dnn.PersonaBar.Library.Helper;
+using Dnn.PersonaBar.Library.Prompt.Models;
+
+namespace Dnn.PersonaBar.Pages.Tests
+{
+    [TestFixture]
+    public class RestorePageUnitTests
+    {
+        TabInfo _tab;
+        string _message;
+        PortalSettings _portalSettings;
+        IConsoleCommand _restorePage;
+
+        Mock<ITabController> _tabControllerMock;
+        Mock<IRecyclebinController> _recyclebinControllerMock;
+        Mock<IContentVerifier> _contentVerifierMock;
+        
+        int _tabId = 91;
+        int _testPortalId = 1;
+
+        [SetUp]
+        public void RunBeforeAnyTest()
+        {
+            _message = string.Empty;
+
+            _tab = new TabInfo();
+            _tab.TabID = _tabId;
+            _tab.PortalID = _testPortalId;
+
+            _portalSettings = new PortalSettings();
+            _portalSettings.PortalId = _testPortalId;
+
+            _tabControllerMock = new Mock<ITabController>();
+            _recyclebinControllerMock = new Mock<IRecyclebinController>();
+            _contentVerifierMock = new Mock<IContentVerifier>();
+        }
+
+        [Test]
+        public void Run_RestorePage_WithValidCommand_ShouldReturnSuccessResponse()
+        {
+            // Arrange            
+            _tabControllerMock.SetReturnsDefault(_tab);
+            _contentVerifierMock.SetReturnsDefault(true);
+            _recyclebinControllerMock.Setup(r => r.RestoreTab(_tab, out _message));
+
+            SetupCommand();
```

Comment:
Why not run `SetupCommand` within `RunBeforeAnyTest`?

Reply:
Its failing for some cases so undo that